src/python/zquantum/core/distribution/moment_based_distance.py

```diff
@@ -0,0 +1,55 @@
+import math
+from typing import TYPE_CHECKING, Dict
+
+if TYPE_CHECKING:
+    from zquantum.core.distribution import MeasurementOutcomeDistribution
+
+
+def compute_moment_based_distance(
+    target_distribution: "MeasurementOutcomeDistribution",
+    measured_distribution: "MeasurementOutcomeDistribution"
+) -> float:
+    """Compute the total variation distance between two distributions,
```

Comment:
Is this comment the same as in the TVD? This is important as "moment based distance" is not quickly googleable. So we want to be precise here.